core/pom.xml

```diff
@@ -114,6 +118,7 @@
         <groupId>org.apache.felix</groupId>
         <artifactId>maven-bundle-plugin</artifactId>
         <configuration>
+          <scrLocation>${src.osgi.location}</scrLocation>
```

Comment:
This and the related property changes actually has no effect because this setting controls where SCR (ie. declarative services) files are generated and not where the source is loaded from. In fact the bundle plugin analyses the bytecode to generate OSGi metadata and doesn't scan the Java source.

The reason the aopalliance import is appearing in the 'no-aop' jar is because bundleplugin 3.5.0 attempts to merge in headers from the main jar into forked builds, like the 'no_aop' munged build. This overwrites the custom 'no_aop' headers.

So the key difference is your change to the `munge-maven-plugin` phase on line 184 - changing that to `generate-sources` means the bundleplugin is no longer merging in those headers. I have an earlier pending PR that solved this a different way: https://github.com/google/guice/pull/1173 - but both solutions are valid, just drop the SCR related changes here because they aren't necessary

Reply:
Damn, I was so focused on how configure the maven-bundle-plugin with another source directory that I've read `srcLocation` instead of `scrLocation` :/ Thanks for the review!
I'll update the PR and let you decide whether you should apply yours or mine. Yours is fixing more stuff.